client/service_create_test.go

```diff
@@ -121,3 +122,94 @@ func TestServiceCreateCompatiblePlatforms(t *testing.T) {
 		t.Fatalf("expected `service_amd64`, got %s", r.ID)
 	}
 }
+
+type pinByDigestTestParams struct {
+	img      string // input image provided by the user
+	expected string // expected image after digest pinning
+}
+
+var dgst = "sha256:c0537ff6a5218ef531ece93d4984efc99bbf3f7497c0a7726c88e2bb7584dc96"
+var pinByDigestTests = []pinByDigestTestParams{
```

Comment:
Small cosmetic request: Can you please move these variables and the type declaration inside the test function, since they are only used by that test?